dpc/src/record/ciphertext.rs

```diff
@@ -133,21 +126,17 @@ impl<N: Network> FromBytes for Ciphertext<N> {
     /// Decode the ciphertext into the ciphertext randomizer, record view key commitment, and record ciphertext.
     #[inline]
     fn read_le<R: Read>(mut reader: R) -> IoResult<Self> {
-        let mut ciphertext = vec![0u8; N::RECORD_CIPHERTEXT_SIZE_IN_BYTES];
-        reader.read_exact(&mut ciphertext)?;
-
         // Decode the ciphertext bytes.
-        let mut cursor = Cursor::new(ciphertext);
-        let ciphertext_randomizer = N::RecordRandomizer::read_le(&mut cursor)?;
-        let record_view_key_commitment = N::RecordViewKeyCommitment::read_le(&mut cursor)?;
+        let ciphertext_randomizer = N::RecordRandomizer::read_le(&mut reader)?;
+        let record_view_key_commitment = N::RecordViewKeyCommitment::read_le(&mut reader)?;
 
         let mut record_bytes = vec![
             0u8;
             N::RECORD_CIPHERTEXT_SIZE_IN_BYTES
                 - N::RecordRandomizer::data_size_in_bytes()
                 - N::RecordViewKeyCommitment::data_size_in_bytes()
         ];
-        cursor.read_exact(&mut record_bytes)?;
+        reader.read_exact(&mut record_bytes)?;
 
```

Comment:
@ljedrz I like that the new logic is more concise. Can you help me build confidence that this ensures the maximal size read is no more than `N::RECORD_CIPHERTEXT_SIZE_IN_BYTES`?

Reply:
The concern here is that we can accept the malformed ciphertexts (by misreading some critical parameters that ultimately do introduce fund vulnerability)

Reply:
A limited size (`N::RECORD_CIPHERTEXT_SIZE_IN_BYTES - N::RecordRandomizer::data_size_in_bytes() - N::RecordViewKeyCommitment::data_size_in_bytes()`) is guaranteed for `record_bytes`, so I'm assuming you're referring to `ciphertext_randomizer` and `record_view_key_commitment` here; this should ideally be constrained in the `FromBytes` impl for `AleoLocator` (the concrete type for `N::RecordRandomizer` and `N::RecordViewKeyCommitment`), but if you want them to additionally be limited here, I can apply [Read::take](https://doc.rust-lang.org/std/io/trait.Read.html#method.take) limiting the reads for `ciphertext_randomizer` and `record_view_key_commitment` to `N::RecordRandomizer::data_size_in_bytes()` and `N::RecordViewKeyCommitment::data_size_in_bytes()` respectively.

Which do you prefer?